lib/es6-promise/promise.js

```diff
@@ -379,5 +379,51 @@ Promise.prototype = {
 */
   catch(onRejection) {
     return this.then(null, onRejection);
+  },
+
+/**
+  `finally` will be invoked regardless of the promise's fate just as native
+  try/catch/finally behaves
+
+  Synchronous example:
+
+  ```js
+  findAuthor() {
+    if (Math.random() > 0.5) {
+      throw new Error();
+    }
+    return new Author();
+  }
+
+  try {
+    return findAuthor(); // succeed or fail
+  } catch(error) {
+    return findOtherAuther();
+  } finally {
+    // always runs
+    // doesn't affect the return value
+  }
+  ```
+
+  Asynchronous example:
+
+  ```js
+  findAuthor().catch(function(reason){
+    return findOtherAuther();
+  }).finally(function(){
+    // author was either found, or not
+  });
+  ```
+
+  @method finally
+  @param {Function} callback
+  @return {Promise}
+*/
+  finally(callback) {
+    let promise = this;
+    let constructor = promise.constructor;
```

Comment:
this needs to use `Symbol.species` to look up the constructor

Reply:
We don't currently support species stuff here.